quic/s2n-quic-transport/src/recovery/manager.rs

```diff
@@ -251,16 +257,27 @@ impl<Config: endpoint::Config> Manager<Config> {
             .on_packet_sent(ecn, path_event!(path, path_id), publisher);
         self.sent_packet_ecn_counts.increment(ecn);
 
-        if outcome.is_congestion_controlled {
-            if outcome.ack_elicitation.is_ack_eliciting() {
-                self.time_of_last_ack_eliciting_packet = Some(time_sent);
-            }
+        if outcome.ack_elicitation.is_ack_eliciting() {
+            self.time_of_last_ack_eliciting_packet = Some(time_sent);
             //= https://www.rfc-editor.org/rfc/rfc9002#section-6.2.1
             //# A sender SHOULD restart its PTO timer every time an ack-eliciting
             //# packet is sent or acknowledged,
-            let is_handshake_confirmed = context.is_handshake_confirmed();
-            let path = context.path_mut_by_id(context.path_id());
-            self.update_pto_timer(path, time_sent, is_handshake_confirmed);
+            self.pto_update_pending = true;
```

Comment:
What do you think about doing a `debug_assert!(!self.pto_update_pending)` inside another method to make sure it gets updated at the end of a burst? I think it should probably work in the `on_timeout` method.

Reply:
Makes sense, I'll add it